app/views/play/VoyagerUnitMap.vue

```diff
@@ -0,0 +1,211 @@
+    TODO: Update after clarification from Robin - i.e. if we want them to be in sync always, so that its easy to update classroom snapshot of levels;
+    or if we want specific properties to only exist in one of these as the single source of truth.
+    */
+    async buildLevelsDataForCourse () {
+      try {
+        const courseInstance = await api.courseInstances.get({courseInstanceID: this.courseInstanceId})
+        const courseId = this.courseId = courseInstance.courseID
+        const classroomId = courseInstance.classroomID
+        
+        // campaign snapshot of the levels
+        const existingCampaignLevels = this.campaignData.levels
+
+        // classroom snapshot of the levels for the course
+        const classroom = this.classroom = await api.classrooms.get({classroomID: classroomId})
+        const classroomCourseLevels =  _.find(classroom.courses, {_id: courseId}).levels
+
+        // get levels data for the levels in the classroom snapshot
+        const classroomCourseLevelsData = await api.classrooms.getCourseLevels({classroomID: classroomId, courseID: courseId})
+
+        for (let level of classroomCourseLevels) {
+          this.classroomLevelMap[level.original] = level
+        }
+        
+        let courseLevelsData = {}
+        for (let level of classroomCourseLevelsData) {
+          let original = level.original
+          if (existingCampaignLevels[original]) {
+            courseLevelsData[original] = existingCampaignLevels[original]
+          }
+          else {
+            // a level which has been removed from the campaign but is saved in the course
+            courseLevelsData[original] = level
+          }
+          // carry over position, nextLevels, first property stored in classroom course, if there are any
+          if (this.classroomLevelMap[original].position)
+            courseLevelsData[original].position = this.classroomLevelMap[original].position
+          if (this.classroomLevelMap[original].nextLevels)
+            courseLevelsData[original].nextLevels = this.classroomLevelMap[original].nextLevels
+          if (this.classroomLevelMap[original].first)
+            courseLevelsData[original].first = this.classroomLevelMap[original].first
+        }
+        // TODO: what if level was removed from campaign, and classroom level map doesnt contain positions, nextLevels, and first? Is it a valid scenario?
+        // If yes, then use the logic from CampaignView for setting the missing positions using the prev and next level positions
+        // Not sure how to set the .first and .nextLevels in that case.
+
+        return courseLevelsData
+      }
+      catch (err) {
+        return Promise.reject(err)
+      }
+    },
+    createLevelStatusMap() {
+      // Remove the level sessions for the levels played in another language - for the classroom version of unit map
+      // TODO: What if played in same language in another classroom
```

Comment:
Is this TODO something we are discussing solutions for? Or waiting for more info? Thanks!

Reply:
This is an edge case that we didnt handle as of now. I am planning to clean up the TODO comments, and move them to asana/eng doc.